lxd/auth/generate/main.go

```diff
@@ -0,0 +1,256 @@
+	var builder strings.Builder
+
+	// Comment that the file is generated.
+	builder.WriteString("// Code generated by `make update-auth`. DO NOT EDIT.\n\n")
+	builder.WriteString("package auth\n\n")
+
+	// Imports.
+	builder.WriteString("import (\n")
+	builder.WriteString("\t\"github.com/canonical/lxd/shared/entity\"\n")
+	builder.WriteString(")\n\n")
+
+	// Entitlement type definition.
+	builder.WriteString("// Entitlement is a representation of the relations that group members can have with entity types.\n")
+	builder.WriteString("type Entitlement string\n\n")
+
+	// Write a list of all entitlements.
+	builder.WriteString("const (\n")
+	for i, entitlement := range allEntitlements {
+		// For each entitlement, get the entity types that it can be applied to (for use in comment).
+		var entityTypes []string
+		for entityType, entitlements := range entityToEntitlements {
+			for _, e := range entitlements {
+				if entitlement.relation == e.relation {
+					entityTypes = append(entityTypes, string(entityType))
+					break
+				}
+			}
+		}
+
+		for i := range entityTypes {
+			entityTypes[i] = fmt.Sprintf("entity.Type%s", snakeToPascal(entityTypes[i]))
+		}
+
+		sort.Strings(entityTypes)
+
+		builder.WriteString(fmt.Sprintf("\t// Entitlement%s is the \"%s\" entitlement. It applies to the following entities: %s.\n", snakeToPascal(entitlement.relation), entitlement.relation, strings.Join(entityTypes, ", ")))
+
+		if i == len(allEntitlements)-1 {
+			builder.WriteString(fmt.Sprintf("\tEntitlement%s Entitlement = \"%s\"\n", snakeToPascal(entitlement.relation), entitlement.relation))
+		} else {
+			builder.WriteString(fmt.Sprintf("\tEntitlement%s Entitlement = \"%s\"\n\n", snakeToPascal(entitlement.relation), entitlement.relation))
+		}
+	}
+
+	builder.WriteString(")\n\n")
+
+	// To ensure the entity to entitlement map is always in the same order, get a list of entity types and sort it alphabetically.
+	var entityTypes []string
+	for entityType := range entityToEntitlements {
+		entityTypes = append(entityTypes, string(entityType))
+	}
+
+	sort.Strings(entityTypes)
+
+	// Map of entity.Type to slice of entitlements.
+	builder.WriteString("var entityTypeToEntitlements = map[entity.Type][]Entitlement{\n")
+	for _, entityType := range entityTypes {
+		entitlements := entityToEntitlements[entity.Type(entityType)]
+		builder.WriteString(fmt.Sprintf("\tentity.Type%s: {\n", snakeToPascal(entityType)))
+		for _, entitlement := range entitlements {
+			// Here we can add the comment from the OpenFGA model.
+			builder.WriteString(fmt.Sprintf("\t\t// %s\n", entitlement.description))
+			builder.WriteString(fmt.Sprintf("\t\tEntitlement%s,\n", snakeToPascal(entitlement.relation)))
+		}
+
+		builder.WriteString("\t},\n")
+	}
+
+	builder.WriteString("}\n")
+
+	// Replace incorrect `entity.TypeGroup` with `entity.TypeAuthGroup`.
+	s := strings.Replace(builder.String(), "entity.TypeGroup", "entity.TypeAuthGroup", -1)
```

Comment:
This stood out as odd to me - why is this needed and please can you add a comment explaining why to future readers.